seatunnel-engine/seatunnel-engine-storage/imap-storage-file/src/main/java/org/apache/seatunnel/engine/imap/storage/file/IMapFileStorage.java

```diff
@@ -0,0 +1,442 @@
+import org.apache.seatunnel.engine.imap.storage.file.bean.IMapFileData;
+import org.apache.seatunnel.engine.imap.storage.file.common.FileConstants;
+import org.apache.seatunnel.engine.imap.storage.file.disruptor.WALDisruptor;
+import org.apache.seatunnel.engine.imap.storage.file.disruptor.WALEventType;
+import org.apache.seatunnel.engine.imap.storage.file.future.RequestFuture;
+import org.apache.seatunnel.engine.imap.storage.file.future.RequestFutureCache;
+import org.apache.seatunnel.engine.imap.storage.file.orc.OrcReader;
+import org.apache.seatunnel.engine.imap.storage.file.scheduler.ArchiveFileTask;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ClassUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hadoop.mapred.JobConf;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * IMapFileStorage
+ * Please notice :
+ * Only applicable to big data (kv) storage. Otherwise, there may be a lot of fragmented files
+ * This is not suitable for frequently updated scenarios because all data is stored as an appended file.
+ * There is no guarantee that all files will be up-to-date when a query is made, and this delay depends on the archive cycle.
+ * If you write large amounts of data in batches, it is best to archive immediately.
+ * Some design detail:
+ * base on file, use orc file to store data
+ * use disruptor to write data to file
+ * use orc reader to read data from file
+ * use wal to ensure data consistency
+ * use request future to ensure data consistency
+ */
+@Slf4j
+public class IMapFileStorage implements IMapStorage {
+
+    public FileSystem fs;
+
+    public String namespace;
+
+    /**
+     * virtual region, Randomly generate a region name
+     */
+    public String region;
+
+    /**
+     * like OSS bucket name
+     * It is used to distinguish data storage locations of different business.
+     */
+    public String businessName;
+
+    /**
+     * This parameter is primarily used for cluster isolation
+     * we can use this to distinguish different cluster, like cluster1, cluster2
+     * and this is also used to distinguish different business
+     */
+    public String clusterName;
+
+    /**
+     * We used disruptor to implement the asynchronous write.
+     */
+    WALDisruptor walDisruptor;
+
+    /**
+     * serializer, default is ProtoStuffSerializer
+     */
+    Serializer serializer;
+
+    private String businessRootPath = null;
+
+    public static final int DEFAULT_ARCHIVE_WAIT_TIME_MILLISECONDS = 1000 * 60;
```

Comment:
Can `DEFAULT_ARCHIVE_WAIT_TIME_MILLISECONDS` be set to 0 manually, Most usage scenarios need to write files synchronously